### PR TITLE
gpu: Add NVIDIA GPU Confidential kernel target

### PR DESCRIPTION
**Pull request by zvonkok**
This is a follow up to the work of minimizing targets, unifying TDX,SNP builds for NVIDIA GPUs

Fixes: #8828

Follow-up PR will remove the old targets.

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, thanks @zvonkok!

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh [hunk 1]
```diff
@@ -364,6 +363,16 @@ install_kernel_nvidia_gpu() {
 		"-g nvidia -u ${kernel_url} -H deb"
 }
 
+#Instal GPU and TEE enabled kernel asset
```
Comment (stevenhorsman) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:366]: minor nit: typo
```suggestion
#Install GPU and TEE enabled kernel asset(s)
```

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks!

**Review by GabyCT — APPROVED** [review]

**Review by lifupan — APPROVED** [review]

**Comment by zvonkok** [comment]
FYI @cdesiniotis @tariq1890 @shivamerla

**Pull request merged**